backend/src/order.ts

```diff
@@ -318,9 +318,9 @@ async function getPaymentHandler(order_id: number, provider?: string): Promise<I
     log.log('debug', 'No provider given, acquiring one from database');
     try {
       const res = await db.query('select payment_provider from nk2_orders where id = :order_id', {order_id});
-      p = res[0].provider;
-
-      if (p === null) throw "No provider assigned to order";
+      p = res[0].payment_provider;
+      log.log('debug', 'Got provider', {order_id, payment_provider: p})
+      if (p === null || p === undefined) throw "No provider assigned to order";
```

Comment:
Could just be if(!p) right? Other falsy values shouldn't be valid providers anyways

Reply:
Yes, fixed